### DISCUSSION TITLE
How to open preline modal programmatically?

### DISCUSSION THREAD

**Discussion by Aux9** [htmlstreamofficial/preline#519] [General]
Hi..
lest say I'm using one of the modals [here](https://preline.co/examples/overlays-modals.html)...

How can I open and close it without the button? I mean using JavaScript directly?

**Comment by P00HB33R** [comment]
Using javascript:

import { HSOverlay } from 'preline';

function myFunc(){

//Open
HSOverlay.open('#hs-modal-id-you-want-to-open');

//Closing
HSOverlay.close('#hs-modal-id-you-want-to-open');
}

async function myAsyncFunc(){

//Open
await HSOverlay.open('#hs-modal-id-you-want-to-open');

//Closing
await HSOverlay.close('#hs-modal-id-you-want-to-open');
}

**Reply by elasawilmar7** [reply]
how to open modal with a function in livewire component? [with this example ](https://preline.co/docs/modal.html#basic-usage)

**Comment by steffenr** [comment]
I had a similar issue and it can be solved by using the following snippet (for vanilla javascript). 

```
<script>
    window.addEventListener('load', () => {
        HSOverlay.open('#hs-full-screen-modal-below-md');            
    });
</script>
```
